src/components/Interpretations/common/getInterpretationAccess.js

```diff
@@ -1,9 +1,23 @@
-const isCreatorOrSuperuser = (object, currentUser) =>
-    object?.createdBy.id === currentUser?.id ||
-    currentUser?.authorities.has('ALL')
+// For backwards compatibility
+// accept both Set (from the old d2.currentUser object) and array
+const hasAuthority = (authorities = [], authority) => {
+    if (Array.isArray(authorities)) {
+        return authorities.includes(authority)
+    }
+    return typeof authorities.has === 'function'
+        ? authorities.has(authority)
+        : false
```

Comment:
I'm not sure that returning `false` here is a good idea. This function should receive either an array or a set of authorities. If it doesn't receive that, the function was called in the wrong way and it can't compute the authorities, so should throw an error. Firstly `false` is actually wrong: the user may or may not have the right authority, but we are unable to asses this. Secondly, this will make it harder to debug things, since this way of handling things obscures things that have gone wrong at an earlier stage (i.e. user authorities not being available when the function was called). I think [failing-fast](https://www.martinfowler.com/ieeeSoftware/failFast.pdf) will improve maintainability and debug-ability in the long run.